ts/src/bitmex.ts

```diff
@@ -1486,17 +1556,18 @@ export default class bitmex extends Exchange {
         //         "timestamp": "2019-03-05T12:47:02.762Z"
         //     }
         //
+        const marketId = this.safeString (trade, 'symbol');
+        const symbol = this.safeSymbol (marketId, market);
         const timestamp = this.parse8601 (this.safeString (trade, 'timestamp'));
         const priceString = this.safeString2 (trade, 'avgPx', 'price');
-        const amountString = this.safeString2 (trade, 'size', 'lastQty');
-        const execCost = this.safeString (trade, 'execCost');
-        const costString = Precise.stringDiv (Precise.stringAbs (execCost), '1e8');
+        const amountString = this.convertFromRawQuantity (symbol, this.safeString2 (trade, 'size', 'lastQty'));
+        const execCost = this.numberToString (this.convertFromRawCost (symbol, this.safeString (trade, 'execCost')));
         const id = this.safeString (trade, 'trdMatchID');
         const order = this.safeString (trade, 'orderID');
         const side = this.safeStringLower (trade, 'side');
         // price * amount doesn't work for all symbols (e.g. XBT, ETH)
         let fee = undefined;
-        const feeCostString = Precise.stringDiv (this.safeString (trade, 'execComm'), '1e8');
+        const feeCostString = this.numberToString (this.convertFromRawCost (symbol, this.safeString (trade, 'execComm')));
```

Comment:
what about avoiding a new method? maybe just using `convertFromRawQuantity` there too, but
 `convertFromRawQuantity (symbol, rawQuantity) ` 
signature can be change into:
 `convertFromRawQuantity (symbol, rawQuantity, currencySide = 'base') ` 

and you can just pass the 'quote' in third argument

Reply:
so it would not matter whether we are parsing "cost" or what, we would have a more generic method.

Reply:
done

Reply:
> @ttodua yeah I will dry it in a single method but keep the wrappers (rawCost, rawQuantity) to make things clear

tbh, I am not sure the wrappers are a good idea, maybe they will more complicate thing for understanding than two pure methods that we basically need. it is converting `from exchange-quantiy into unified-quantity`, and converting `from unified quantity to exchange-quantity`.

Reply:
@ttodua why is it more complicated? Tomorrow if someone else tries to parse a cost on a different method I think it's easier to have a `parseFromRawCost` method than the "generic" version of it where you have to mention the currency side.

Reply:
because to follow that logic, then we should create `convertFromRawSize, convertFromRawNotional,  convertFromRawOhlcv, and etcetera` object-name specific method.  
now we have:
```
convertFromRealAmount
convertToRealAmount
convertFromQuantity //  need name change, somewhat semantically same as convertToRealAmount
convertFromRawQuantity // maybe we can remove this at all, why do we need this?
convertFromRawCost 
```

maybe we rename `convertFromQuantity` to `convertToRealAmount**ByMarket**` (or somewhat similar, because `convertFromQuantity` is just an extra wrapper of  `convertToRealAmount`

Reply:
another argument against adding nested `convertFromOBJECTSPECIFIC` is that, say alone `convertFromRawCost` might not always be based on `quote` (because of inverse markets) so it's an extra complicated layer imo and pure & explicit `convertToRealAmount` (with passing a market 'base' or 'quote' currencycode) looks more clear to me.
but as you wish.

Reply:
@ttodua 

> convertFromRawSize, convertFromRawNotional,  convertFromRawOhlcv, and etceter

No we don't need this level of specificity, bitmex basically only has two "different" scales, one for the amounts (order amount, volume, etc) and the other for the costs, for instance price is the real one. 

> maybe we rename convertFromQuantity to convertToRealAmount**ByMarket** (or somewhat similar, because convertFromQuantity is just an extra wrapper of convertToRealAmount

ok I removed `convertfromQuantity` kept `convertFromRawQuantity` and ` convertFromRawCost`

Reply:
that's what I doubt in -  bitmex does not have two different scales, it has only one scale which depends on currency. we make two different scales by imposing the conversion through `market['base']` or `market['quote']` but either 'base' or 'quote' are just currency-codes, which are then sent to `convertFromRealAmount (currency_code` which does that conversion. but as you consider ofc.
also, you might check if `quote` is correct side for inverse or quanto market's cost parsing.

Reply:
> t has only one scale which depends on currency.

Yes I agree probably I phrased it wrong, but I think we're overcomplicating it, right now `convertFromRawCost` is literally a one-liner that wraps `convertFromRawQuantity` , so what is wrong with it? Upon editing bitmex you can always use one or another